docs/proposals/scheduling/20221227-cpu-static-reservation.md

```diff
@@ -0,0 +1,176 @@
+
+### Implementation Details
+By adding annotation to the node, we specify the resources to be reserved by the koordinator, such as memory, cpu, etc. For cpu, we can specify the total amount of cpu to be reserved, or we can specify which cores to explicitly reserve. 
+```golang
+// resources that koordinator reserved, and you can reserve other resource if needed.
+type KoordReserved struct {
+	// total cpus need to be reserved, just like "2C"
+	CPUNums      string `json:"CPUNums"`
+	// specific cpus need to be reserved, just like "0-6", it means that 0-6 core cannot be used by k8s.
+	specificCPUs string `json:"specificCPUs"`
+	// eg: "1Gi"
+	memory       string `json:"memory"`
+}
+```
+#### koordlet   
+1. When koordlet reports `NodeResourceTopology`, it updates the explicitly reserved cores in node.annntation to `NodeResourceTopology.annotation["node.koordinator.sh/reserved-cpus"]`, so that the `runtimehook` can ignore these cores when fetching the shared-pool from nodetopo.
+
+2. cpu supress:  
+it will evict some pod whose qos is not lse. we should ignore the cpus reserved by node when supress with cpuset policy.  
+
+```
+cpuset(lsr) = processor(lsr) - cpusReservedByNode
+cpuset(!lse/lser) = processor(!lse/lser) - cpusReservedByNode
+``` 
+
+3. runtimehooks:  
+it will ignored the cpus reserved by `node.annotations` when create container. 
+```
+cpus(allocatable) = cpus(shared-pools) - cpus(reservedByNode)
+```  
+  
+<font color=Chocolate>*attention:*</font> The cpu occupied by the lse pod on the node is not considered here for now, but it will be removed later in the cpuset suppression phase.
+
+
+#### koord-manager  
+should update batch-resource on node, just like `batch-memory`,`batch-cpu`.the implementation of this part depends on the `nodeResource`  controller in koord-manager, so for batch-resource, the `batchResourceFit` plugin can take into account the resources reserved by the node.
+
+```
+Node(BE).Alloc = Node.Total - Node.Reserved - node.anno.reserved - System.Used - Pod(LS).Used
+```
+
+#### koord-scheduler
+
+- for memory:  
+  node.alloc - node.requested.memory - node.annotation[koord-reserved].memory >= pod.request
+- for cpu:  
+  1. both nums and specific are declared on node.anno:   
+     nums will be ignored.
+  2. only nums:  
+     node.alloc.cpu - node.requested.cpu - node.annotation[koord-reserved].cpu >= pod.request.cpu
+  3. only specific cpus:  
+     pod.anno[specific-cpus] $\bigcap$ node.anno[koord-specific-cpus-reserved] = $\phi$   
+    
+  *[description]*: We can only get the container's cpuset from two places, one is `pod.anno` and the other is `nodetopo.shared-pool`. If a pod does not declare the specific cores to be used, it only needs to ensure that there are indeed enough resources on the node to run the container; if it is already scheduled, then for the cores explicitly reserved on the node, these cores should not be included in the range of cpuset that the container can use.
+
+...
```

Comment:
If the reserved CPU set is updated, what is the process and whether there is a conflict

Reply:
this is indeed a problem, eviction should be considered in this situation.